src/fsharp/ConstraintSolver.fs

```diff
@@ -2121,7 +2121,9 @@ and ReportNoCandidatesError (csenv:ConstraintSolverEnv) (nUnnamedCallerArgs, nNa
     // Many methods of different arities, all incorrect 
     | _, _, ([], (cmeth :: _)), _, _ -> 
         let minfo = cmeth.Method
```

Comment:
It looks like it would be extra work to list all found overloads. Is it feasible to have a `NicePrint.formatMethoInfosToBufferFreeStyle` or appropriate?

Reply:
@baronfel as you can see, all found overloads are available as a list, we use the first one. Showing all of them would be problematic because there are methods with huge number of overloads. A good option would be to show, say, 5 of them max, separated with new line.

On the other hand, other error messages, that are raised in nearby code may be improved as well...

Reply:
Talking with @KevinRansom , we think printing one method solves 99% of most developers' problems. Printing more out may be an overload on everyone's brains, no pun intended. :)

Reply:
But showing all the overloads gives the maximally quick way to choose a right one.

Reply:
When I'm calling methods, I always rely on the tooling to help me choose the right one, never the error message.

Reply:
![image](https://user-images.githubusercontent.com/873919/39066758-d76e7668-44de-11e8-86ec-5eb07ee605c8.png)

I think it's pretty cool.

Reply:
Yes, it is. @KevinRansom what do you think?

Reply:
Another one

![image](https://user-images.githubusercontent.com/873919/39066908-729e9ad2-44df-11e8-93ed-413bc7d810e6.png)

Reply:
@vasily-kirichenko 

This is a bit like the work Steffen did a while ago to improve error messages, to I believe make our error messages more Elm like by proposing a list of options.  I remember at the time thinking, I would hate it, and now I love it.  So you are probably right that displaying the complete list would be beneficial.  Let's see what it looks like in actual use.

Reply:
@vasily-kirichenko how does it look for many overloads? I found a few examples of methods with large amount of overloads in the following locations: 

```
(Member, Count, DeclaringType)
("WriteLine", 18, System.CodeDom.Compiler.IndentedTextWriter)
("Replace", 12, System.Text.RegularExpressions.Regex)
("WriteEntry", 10, System.Diagnostics.EventLog)
("AddRange", 8, System.Net.HttpWebRequest)
("Send", 8, System.Net.Sockets.Socket)
("Receive", 8, System.Net.Sockets.Socket)
("SendAsync", 8, System.Net.NetworkInformation.Ping)
("SendPingAsync", 8, System.Net.NetworkInformation.Ping)
("ToString", 7, System.ComponentModel.MaskedTextProvider)
("Complete", 7, System.Net.Sockets.SocketAsyncResult)
("SignalAndWait", 6, System.Threading.Barrier)
("Match", 6, System.Text.RegularExpressions.Regex)
("Split", 6, System.Text.RegularExpressions.Regex)
("Start", 6, System.Diagnostics.Process)
("GetEvents", 6, System.ComponentModel.TypeDescriptor)
("GetProperties", 6, System.ComponentModel.TypeDescriptor)
("Import", 6, System.Security.Cryptography.X509Certificates.X509Certificate2)
("Write", 6, System.Diagnostics.Tracing.EventSource)
```

Reply:
@KevinRansom yeah, I think this is one of the @forki improvements:

![image](https://user-images.githubusercontent.com/873919/39068913-a51ecc1e-44e6-11e8-9c6f-4f425c483dc7.png)

I don't like it's all comma separated, in one line though.

Reply:
@baronfel Regex:

![image](https://user-images.githubusercontent.com/873919/39069012-fc6135ca-44e6-11e8-916c-b487e3b7b7df.png)

Looks ok to me.

Reply:
In Error List

![image](https://user-images.githubusercontent.com/873919/39069061-364e33be-44e7-11e8-9f15-7cdafd565559.png)